py_gtfs_rt_ingestion/py_gtfs_rt_ingestion/convert_gtfs_rt.py

```diff
@@ -0,0 +1,162 @@
+import json
+import logging
+import os
+
+from datetime import datetime
+from typing import Optional
+from concurrent.futures import ThreadPoolExecutor
+
+import pyarrow
+from pyarrow import fs
+
+from .error import NoImplException
+from .converter import Converter, ConfigType
+from .gtfs_rt_detail import GTFSRTDetail
+
+from .config_rt_alerts import RtAlertsDetail
+from .config_rt_bus_vehicle import RtBusVehicleDetail
+from .config_rt_bus_trip import RtBusTripDetail
+from .config_rt_trip import RtTripDetail
+from .config_rt_vehicle import RtVehicleDetail
+
+
+POOL_SIZE = 4
+
+
+class GTFSRTConverter(Converter):
+    """
+    Converter that handles GTFS Real Time JSON data
+
+    https_cdn.mbta.com_realtime_Alerts_enhanced.json.gz
+    https_cdn.mbta.com_realtime_TripUpdates_enhanced.json.gz
+    https_cdn.mbta.com_realtime_VehiclePositions_enhanced.json.gz
+    https_mbta_busloc_s3.s3.amazonaws.com_prod_TripUpdates_enhanced.json.gz
+    https_mbta_busloc_s3.s3.amazonaws.com_prod_VehiclePositions_enhanced.json.gz
+    https_mbta_integration.mybluemix.net_vehicleCount.gz
+    """
+
+    def __init__(self, config_type: ConfigType) -> None:
+        Converter.__init__(self, config_type)
+
+        # Depending on filename, assign self.details to correct implementation
+        # of GTFSRTDetail class.
+        self.detail: GTFSRTDetail
+
+        if config_type == ConfigType.RT_ALERTS:
+            self.detail = RtAlertsDetail()
+        elif config_type == ConfigType.RT_TRIP_UPDATES:
+            self.detail = RtTripDetail()
+        elif config_type == ConfigType.RT_VEHICLE_POSITIONS:
+            self.detail = RtVehicleDetail()
+        elif config_type == ConfigType.BUS_VEHICLE_POSITIONS:
+            self.detail = RtBusVehicleDetail()
+        elif config_type == ConfigType.BUS_TRIP_UPDATES:
+            self.detail = RtBusTripDetail()
+        else:
+            raise NoImplException(f"No Specialization for {config_type}")
+
+    def convert(self, files: list[str]) -> list[tuple[str, pyarrow.Table]]:
+        pa_table = pyarrow.table(
+            self.detail.empty_table(), schema=self.detail.export_schema
+        )
+
+        logging.info(
+            "Creating pool with %d threads, %d cores available",
+            POOL_SIZE,
+            os.cpu_count(),
+        )
+
+        with ThreadPoolExecutor(max_workers=POOL_SIZE) as executor:
+            for result in executor.map(self.gz_to_pyarrow, files):
+                if result is not None:
+                    pa_table = pyarrow.concat_tables([pa_table, result])
```

Comment:
threading moved from ingest to here. now that it lives here, we can simplify the map and have an easier time moving files into archive or error lists inside of the `gz_to_pyarrow` method.